x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/create_job.ts

```diff
@@ -0,0 +1,319 @@
+  startString: string,
+  endString: string,
+  query: Query,
+  filters: Filter[],
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient,
+  timeFilter: TimefilterContract
+) {
+  try {
+    const { jobConfig, datafeedConfig, createdBy } = await createADJobFromLensSavedObject(
+      vis,
+      query,
+      filters,
+      dataViewClient,
+      kibanaConfig
+    );
+
+    let start: number | undefined;
+    let end: number | undefined;
+    let includeTimeRange = true;
+
+    try {
+      // attempt to parse the start and end dates.
+      // if start and end values cannot be determined
+      // instruct the job cloning code to auto-select the
+      // full time range for the index.
+      const { min, max } = timeFilter.calculateBounds({ to: endString, from: startString });
+      start = min?.valueOf();
+      end = max?.valueOf();
+
+      if (start === undefined || end === undefined || isNaN(start) || isNaN(end)) {
+        throw Error('Incompatible time range');
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      includeTimeRange = false;
+      start = undefined;
+      end = undefined;
+    }
+
+    // add job config and start and end dates to the
+    // job cloning stash, so they can be used
+    // by the new job wizards
+    stashJobForCloning(
+      {
+        jobConfig,
+        datafeedConfig,
+        createdBy,
+        start,
+        end,
+      } as JobCreatorType,
+      true,
+      includeTimeRange,
+      !includeTimeRange
+    );
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+  }
+}
+
+export async function canCreateADJob(
+  vis: LensSavedObjectAttributes,
+  query: Query | undefined,
+  filters: Filter[] | undefined,
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient
+) {
+  try {
+    const jobItems = await createADJobFromLensSavedObject(
+      vis,
+      query ?? getDefaultQuery(),
+      filters ?? [],
+      dataViewClient,
+      kibanaConfig
+    );
+
+    return !!jobItems;
+  } catch (error) {
+    return false;
+  }
+}
+
+async function createADJobFromLensSavedObject(
+  vis: LensSavedObjectAttributes,
+  query: Query,
+  filters: Filter[],
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient
+) {
+  const { fields, timeField, splitField, dataView } = await extractFields(vis, dataViewClient);
+
+  const jobConfig = createEmptyJob();
+  const datafeedConfig = createEmptyDatafeed(dataView.title);
+
+  const combinedFiltersAndQueries = combineQueriesAndFilters(
+    { query, filters },
+    { query: vis.state.query, filters: vis.state.filters },
+    dataView,
+    kibanaConfig
+  );
+
+  datafeedConfig.query = combinedFiltersAndQueries;
+
+  jobConfig.analysis_config.detectors = createDetectors(fields, splitField);
+
+  jobConfig.data_description.time_field = timeField.sourceField;
+  jobConfig.analysis_config.bucket_span = '15m';
```

Comment:
Can we use DEFAULT_BUCKET_SPAN here?

Reply:
Updated in https://github.com/elastic/kibana/pull/129762/commits/b35ac1e50428eebff8b83389d167fa6cf02af651